src/lib/models/outoforder/Core.cc

```diff
@@ -16,77 +16,92 @@ namespace outoforder {
-      issuePorts_(config["Execution-Units"].size(), {1, nullptr}),
-      completionSlots_(
-          config["Execution-Units"].size() +
-              config["Pipeline-Widths"]["LSQ-Completion"].as<unsigned int>(),
-          {1, nullptr}),
+          config::SimInfo::getValue<int>(config["Pipeline-Widths"]["FrontEnd"]),
+          nullptr),
+      issuePorts_(config["Execution-Units"].num_children(), {1, nullptr}),
+      completionSlots_(config["Execution-Units"].num_children() +
+                           config::SimInfo::getValue<int>(
+                               config["Pipeline-Widths"]["LSQ-Completion"]),
+                       {1, nullptr}),
       loadStoreQueue_(
-          config["Queue-Sizes"]["Load"].as<unsigned int>(),
-          config["Queue-Sizes"]["Store"].as<unsigned int>(), dataMemory,
-          {completionSlots_.data() + config["Execution-Units"].size(),
-           config["Pipeline-Widths"]["LSQ-Completion"].as<unsigned int>()},
+          config::SimInfo::getValue<uint32_t>(config["Queue-Sizes"]["Load"]),
+          config::SimInfo::getValue<uint32_t>(config["Queue-Sizes"]["Store"]),
+          dataMemory,
+          {completionSlots_.data() + config["Execution-Units"].num_children(),
+           config::SimInfo::getValue<size_t>(
+               config["Pipeline-Widths"]["LSQ-Completion"])},
           [this](auto regs, auto values) {
             dispatchIssueUnit_.forwardOperands(regs, values);
           },
           [](auto uop) { uop->setCommitReady(); },
-          config["LSQ-L1-Interface"]["Exclusive"].as<bool>(),
-          config["LSQ-L1-Interface"]["Load-Bandwidth"].as<uint16_t>(),
-          config["LSQ-L1-Interface"]["Store-Bandwidth"].as<uint16_t>(),
-          config["LSQ-L1-Interface"]["Permitted-Requests-Per-Cycle"]
-              .as<uint16_t>(),
-          config["LSQ-L1-Interface"]["Permitted-Loads-Per-Cycle"]
-              .as<uint16_t>(),
-          config["LSQ-L1-Interface"]["Permitted-Stores-Per-Cycle"]
-              .as<uint16_t>()),
+          config::SimInfo::getValue<bool>(
+              config["LSQ-L1-Interface"]["Exclusive"]),
+          config::SimInfo::getValue<uint16_t>(
+              config["LSQ-L1-Interface"]["Load-Bandwidth"]),
+          config::SimInfo::getValue<uint16_t>(
+              config["LSQ-L1-Interface"]["Store-Bandwidth"]),
+          config::SimInfo::getValue<uint16_t>(
+              config["LSQ-L1-Interface"]["Permitted-Requests-Per-Cycle"]),
+          config::SimInfo::getValue<uint16_t>(
+              config["LSQ-L1-Interface"]["Permitted-Loads-Per-Cycle"]),
+          config::SimInfo::getValue<uint16_t>(
+              config["LSQ-L1-Interface"]["Permitted-Stores-Per-Cycle"])),
       fetchUnit_(fetchToDecodeBuffer_, instructionMemory, processMemorySize,
-                 entryPoint, config["Fetch"]["Fetch-Block-Size"].as<uint16_t>(),
+                 entryPoint,
+                 config::SimInfo::getValue<uint32_t>(
```

Comment:
Do we need to go from uint16_t to uint32_t?

Reply:
Miscalculated bit needs, don't need 17 so can narrow to 16bits